be/src/runtime/snapshot_loader.cpp

```diff
@@ -426,61 +433,81 @@ Status SnapshotHttpDownloader::_link_same_rowset_files() {
         remote_rowset_metas.insert({rowset_meta.rowset_id_v2(), rowset_meta});
     }
 
-    for (const auto& local_rowset_meta : local_tablet_meta.rs_metas()) {
-        if (local_rowset_meta.has_resource_id() || !local_rowset_meta.has_source_rowset_id()) {
+    std::unordered_map<std::string, const RowsetMetaPB&> local_rowset_metas;
+    for (const auto& rowset_meta : local_tablet_meta.rs_metas()) {
+        if (rowset_meta.has_resource_id()) { // skip local rowset
+            continue;
+        }
+        local_rowset_metas.insert({rowset_meta.rowset_id_v2(), rowset_meta});
+    }
+
+    RETURN_IF_ERROR(_link_rowset_if_exists_from_origin_to_target<LocalFileStat>(
+            local_tablet_meta, remote_rowset_metas, _local_files, _local_path));
+    RETURN_IF_ERROR(_link_rowset_if_exists_from_origin_to_target<RemoteFileStat>(
+            remote_tablet_meta, local_rowset_metas, _remote_files, _remote_path));
+    return Status::OK();
```

Comment:
This function aims to link the local files as downloaded remote files.

The old logic: link if the local file is a remote file copy.  => local_rowset.source_rowset_id() == remote_rowset.rowset_id()
link $local_file to $remote_file

Now, we need to add a new logic: link if the remote file is a local file copy.  => remote_rowset.source_rowset_id() == local_rowset.rowset_id().
link $local_file to $remote_file

The only different thing is the condition, and the link progress are the same. So this change is wrong.